aws/resource_aws_emr_instance_fleet_test.go

```diff
@@ -0,0 +1,551 @@
+            "kinesis:DescribeStream",
+            "kinesis:GetRecords",
+            "kinesis:GetShardIterator",
+            "kinesis:MergeShards",
+            "kinesis:PutRecord",
+            "kinesis:SplitShard",
+            "rds:Describe*",
+            "s3:*",
+            "sdb:*",
+            "sns:*",
+            "sqs:*"
+        ]
+    }]
+}
+EOT
+}
+resource "aws_emr_cluster" "test" {
+  name                 = "%[1]s"
+  release_label        = "emr-5.30.1"
+  applications         = ["Hadoop", "Hive"]
+  log_uri              = "s3n://terraform/testlog/"
+  master_instance_fleet    {
+    instance_type_configs        {
+          instance_type = "m3.xlarge"
+        }
+      target_on_demand_capacity = 1
+    }
+  core_instance_fleet {
+    instance_type_configs {
+      bid_price_as_percentage_of_on_demand_price = 100
+      ebs_config {
+        size                 = 100
+        type                 = "gp2"
+        volumes_per_instance = 1
+      }
+      instance_type     = "m4.xlarge"
+      weighted_capacity = 1
+    }
+    name                      = "core fleet"
+    target_on_demand_capacity = 1
+    target_spot_capacity      = 0
+  }
+  service_role = aws_iam_role.emr_service.arn
+  depends_on   = [
+    aws_route_table_association.test,
+    aws_iam_role_policy_attachment.emr_service,
+    aws_iam_role_policy_attachment.emr_instance_profile,
+  ]
+  ec2_attributes {
+    subnet_id                         = aws_subnet.test.id
+    emr_managed_master_security_group = aws_security_group.test.id
+    emr_managed_slave_security_group  = aws_security_group.test.id
+    instance_profile                  = aws_iam_instance_profile.emr_instance_profile.arn
+  }
+}
+`
+
+func testAccAWSEmrInstanceFleetConfig(r string) string {
+	return fmt.Sprintf(testAccAWSEmrInstanceFleetBase+`
+    resource "aws_emr_instance_fleet" "task" {
+      cluster_id            = "${aws_emr_cluster.test.id}"
+      instance_fleet_type   = "TASK"
+      instance_fleet {
+        instance_type_configs        {
+          instance_type = "m3.xlarge"
+          weighted_capacity = 1
+        }
+        launch_specifications {
+          on_demand_specification {
+            allocation_strategy = "lowest-price"
+          }
+        }
+#        launch_specifications {
+#          spot_specification {
+#            allocation_strategy      = "capacity-optimized"
+#            block_duration_minutes   = 0
+#            timeout_action           = "SWITCH_TO_ON_DEMAND"
+#            timeout_duration_minutes = 10
+#          }
+#        }
+        name                      = "emr_instance_fleet_%[1]s"
+        target_on_demand_capacity = 1
+        target_spot_capacity      = 0
+      }
+    }
+`, r)
+}
+
+func testAccAWSEmrInstanceFleetConfigZeroCount(r string) string {
+	return fmt.Sprintf(testAccAWSEmrInstanceFleetBase+`
+    resource "aws_emr_instance_fleet" "task" {
+      cluster_id            = "${aws_emr_cluster.test.id}"
```

Comment:
Nit: Terraform 0.12 syntax here and below

```suggestion
      cluster_id            = aws_emr_cluster.test.id
```